qiskit/optimization/problems/quadratic_program.py

```diff
@@ -1092,6 +1093,79 @@ def from_ising(self,
         self.minimize(constant=offset, linear=linear_terms, quadratic=quadratic_terms)
         offset -= offset
 
+    def get_feasiblity_info(self, x: Union[List[float], np.ndarray]) \
+            -> Tuple[bool, List[Constraint], List[Variable]]:
+        """Returns whether a solution is feasible or not along with the violations.
+        Args:
+            x: the input result list returned by the optimizer
+        Returns:
+            is_feasible: Whether the solution provided is feasible or not.
+            List[Constraint]: List of constraints which are violated.
+            List[Variable]: List of variables which are violated.
+
+        Raises:
+            QiskitOptimizationError: If the input `x` is not same len as total vars
+        """
+        # if input `x` is not the same len as the total vars, raise an error
+        if len(x) != self.get_num_vars():
+            raise QiskitOptimizationError(
+                'The size of solution `x`: {}, does not match the number of problem variables: {}'
+                .format(len(x), self.get_num_vars())
+            )
+
+        # check whether the input satisfy the bounds of the problem
+        satisfied_variables = {}
+        for i, val in enumerate(x):
+            variable = self.get_variable(i)
+            if variable._lowerbound <= val <= variable._upperbound:
+                satisfied_variables[variable.name] = True
+            else:
+                satisfied_variables[variable.name] = False
+
+        # check whether the input satisfy the constraints of the problem
+        satisfied_constraints = {}
+        for constraint in cast(List[Constraint], self._linear_constraints) + \
+                cast(List[Constraint], self._quadratic_constraints):
+            lhs = constraint.evaluate(x)
+            if constraint.sense == ConstraintSense.LE:
+                satisfied_constraints[constraint.name] = lhs <= constraint.rhs
+            elif constraint.sense == ConstraintSense.GE:
+                satisfied_constraints[constraint.name] = lhs >= constraint.rhs
+            elif constraint.sense == ConstraintSense.EQ:
+                satisfied_constraints[constraint.name] = isclose(lhs, constraint.rhs)
+
+        # create a dict containing only unsatisfied variable(s)/constraint(s)
+        final_dict = {k: v for k, v in {**satisfied_variables, **satisfied_constraints}
+                      .items() if not v}
+
+        # debug
+        logger.debug("Violations: %s", final_dict)
```

Comment:
`final_dict` is created for debug purposes only. Please create it only when debug is enabled or remove this.